src/Symfony/Component/Profiler/Profiler.php

```diff
@@ -0,0 +1,387 @@
+        );
+
+        $response->headers->set('X-Debug-Token', $profile->getToken());
+
+        return $this->profile($profile, $request, $response);
+    }
+
+    /**
+     * Profile a Console Command.
+     *
+     * @param Command        $command
+     * @param InputInterface $input
+     * @param int            $exitCode
+     *
+     * @return ProfileInterface|void
+     */
+    public function profileCommand(Command $command, InputInterface $input, $exitCode)
+    {
+        if (!$this->enabled) {
+            return;
+        }
+
+        $profile = new ConsoleProfile($this->generateToken(), $command->getName(), $input->getArguments(), $input->getOptions(), $exitCode);
+
+        return $this->profile($profile);
+    }
+
+    /**
+     * Collects data.
+     *
+     * @param ProfileInterface $profile
+     *
+     * @return ProfileInterface
+     */
+    public function profile(ProfileInterface $profile, Request $request = null, Response $response = null, \Exception $exception = null)
+    {
+        foreach ($this->collectors as $collector) {
+            $collector->setToken($profile->getToken());
+            if ($collector instanceof RuntimeDataCollectorInterface) {
+                $profile->add($collector->getName(), $collector->collect());
+            } elseif ($collector instanceof LateDataCollectorInterface) {
+                // we need to clone for sub-requests
+                $profile->addCollector(clone $collector);
+            } elseif ($collector instanceof DeprecatedDataCollectorInterface) {
+                @trigger_error(sprintf('%s implementing %s will is deprecated since version 2.8 and support for it will be removed in 3.0.', get_class($collector), 'Symfony\Component\HttpKernel\DataCollector\DataCollectorInterface'), E_USER_DEPRECATED);
+                /** @var DeprecatedDataCollectorInterface $clone */
+                $clone = clone $collector;
+                if (!($clone instanceof DeprecatedLateDataCollectorInterface)) {
+                    if (null !== $request && null !== $response) {
+                        $clone->collect($request, $response, $exception);
+                        // we need to clone for sub-requests
+                        $profile->addCollector($clone);
+                    }
+                } else {
+                    // we need to clone for sub-requests
+                    $profile->addCollector($clone);
+                }
+            }
+        }
+
+        return $profile;
+    }
+
+    private function getTimestamp($value)
+    {
+        if (null === $value || '' == $value) {
+            return;
+        }
+
+        try {
+            $value = new \DateTime(is_numeric($value) ? '@'.$value : $value);
+        } catch (\Exception $e) {
+            return;
+        }
+
+        return $value->getTimestamp();
+    }
+
+    /**
+     * Gets the Collectors associated with this profiler.
+     *
+     * @return array An array of collectors
+     */
+    public function all()
```

Comment:
I don't think that it should be possible to access the collectors in the profiler. What's the use case?